certbot-ci/certbot_integration_tests/certbot_tests/test_main.py

```diff
@@ -896,8 +898,29 @@ def test_preferred_chain(context: IntegrationTestsContext) -> None:
 
         dumped = misc.read_certificate(cert_path)
         assert 'Issuer: CN={}'.format(expected) in dumped, \
-               'Expected chain issuer to be {} when preferring {}'.format(expected, requested)
+            'Expected chain issuer to be {} when preferring {}'.format(expected, requested)
 
         with open(conf_path, 'r') as f:
             assert 'preferred_chain = {}'.format(requested) in f.read(), \
-                   'Expected preferred_chain to be set in renewal config'
+                'Expected preferred_chain to be set in renewal config'
+
+
+def test_certificate_validity(context: IntegrationTestsContext) -> None:
+    """Test that --certificate-validity is requesting certificates of the appropirate validity"""
+    pytest.skip('Pebble needs a new release before we can use this tool in the integration test.')
```

Comment:
FWIW I did test this test with a new version of pebble I manually built and it passed.

Reply:
Although this doesn't work locally on macOS, we do not want to skip this test in CI, which will run against Linux and Windows targets.

To make the integration tests that use Pebble succeed on macOS, I can share the workaround I use. Build Pebble locally on your Mac, and then copy the built Pebble binary to `./certbot-ci/certbot_integration_tests/assets/pebble_v2.3.0_linux-amd64`.  You should then be able to run `tox -e integration` successfully (assuming you have `nginx` installed as well).

Reply:
The problem is that pebble doesn't have a release cut with notAfter implemented.

From what I saw in the code, during the CI testing it uses the releases from GH, right? Might be imagining things.

Reply:
>The problem is that pebble doesn't have a release cut with notAfter implemented.

Are you sure? According to https://github.com/letsencrypt/pebble/commit/2c26e5bac888b5418c53dbea203fbdf9d7516104, the feature is present in release v2.3.1.

Reply:
Ah, I see. Certbot downloads Pebble v2.3.0, it's hardcoded in `pebble_artifacts.py`.

If you want to open a PR to update this file, I will approve and merge it, and we can unblock your work that way.

Reply:
We've merged #9260 for the Pebble thing so you should be able to merge master and unskip the test.